localstack/services/cloudwatch/provider_v2.py

```diff
@@ -233,6 +245,63 @@ def get_metric_data(
             MetricDataResults=formatted_results, NextToken=nxt, Messages=messages
         )
 
+    def set_alarm_state(
+        self,
+        context: RequestContext,
+        alarm_name: AlarmName,
+        state_value: StateValue,
+        state_reason: StateReason,
+        state_reason_data: StateReasonData = None,
+    ) -> None:
+        try:
+            if state_reason_data:
+                json.loads(state_reason_data)
+        except ValueError:
+            raise InvalidParameterValueException(
+                "TODO: check right error message: Json was not correctly formatted"
+            )
+
+        store = self.get_store(context.account_id, context.region)
+        alarm = store.Alarms.get(
+            arns.cloudwatch_alarm_arn(
+                alarm_name, account_id=context.account_id, region_name=context.region
+            )
+        )
+        if not alarm:
+            raise ResourceNotFound()
+
+        old_state = alarm.alarm["StateValue"]
+        if state_value not in ("OK", "ALARM", "INSUFFICIENT_DATA"):
+            raise ValidationError(
+                f"1 validation error detected: Value '{state_value}' at 'stateValue' failed to satisfy constraint: Member must satisfy enum value set: [INSUFFICIENT_DATA, ALARM, OK]"
+            )
+
+        self._update_state(context, alarm, state_value, state_reason, state_reason_data)
+
+        if not alarm.alarm["ActionsEnabled"] or old_state == state_value:
+            return
+        if state_value == "OK":
+            actions = alarm.alarm["OKActions"]
+        elif state_value == "ALARM":
+            actions = alarm.alarm["AlarmActions"]
+        else:
+            actions = alarm.alarm["InsufficientDataActions"]
+        for action in actions:
+            data = arns.parse_arn(action)
+            # test for sns - can this be done in a more generic way?
+            if data["service"] == "sns":
+                service = connect_to.get_client(data["service"])
```

Comment:
NIT: we probably should parse the arn first, and set the `region_name` and `aws_access_key_id` here as well, so we are on the safe side. 
Similar like we do it for the https://github.com/localstack/localstack/blob/master/localstack/services/s3/notifications.py#L543-L546

Reply:
:+1: